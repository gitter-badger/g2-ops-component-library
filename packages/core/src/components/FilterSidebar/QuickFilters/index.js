// @flow
export {QuickFilters} from './quick-filters.component'
