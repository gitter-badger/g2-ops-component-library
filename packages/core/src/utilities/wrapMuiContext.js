// @flow
import type { ComponentType, Node } from 'react'

import React, { Component } from 'react'
import injectTapEventPlugin from 'react-tap-event-plugin'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
import getMuiTheme from 'material-ui/styles/getMuiTheme'
import { initializeIcons } from '@uifabric/icons'

type ThemeType = {
  [string]: any,
}

const muiTheme = getMuiTheme()
const copartBlue = '#1d5ab9'
const niceBlue = '#4a90e2'

export const copartBaseTheme: ThemeType = {
    ...muiTheme,
  appBar: {
    ...muiTheme.appBar,
    color: copartBlue,
  },
  baseTheme: {
    ...muiTheme.baseTheme,
    palette: {
      ...muiTheme.baseTheme.palette,
      pickerHeaderColor: copartBlue,
      primary1Color: copartBlue,
      primary2Color: copartBlue,
    },
  },
  flatButton: {
    ...muiTheme.flatButton,
    primaryTextColor: copartBlue,
  },
  palette: {
    ...muiTheme.palette,
    pickerHeaderColor: copartBlue,
    primary1Color: copartBlue,
    primary2Color: copartBlue,
  },
  datePicker: {
    ...muiTheme.datePicker,
    headerColor: copartBlue,
    color: copartBlue,
    selectColor: copartBlue,
  },
  toggle: {
    ...muiTheme.toggle,
    thumbOnColor: copartBlue,
    trackOnColor: niceBlue,
  },
  timePicker: {
    ...muiTheme.timePicker,
    headerColor: copartBlue,
    accentColor: copartBlue,
    selectColor: copartBlue,
  },
  stepper: {
    ...muiTheme.stepper,
    iconColor: copartBlue,
  },
  tabs: {
    selectedTextColor: 'rgba(38, 166, 91, 1.0)',
    textColor: 'rgba(218, 223, 225, 1.0)',
  },
  inkBar: {
    backgroundColor: '#545A63',
  },
}

initializeIcons()
injectTapEventPlugin()

// a higher order function that provides mui context to the Component being passed.
export function wrapMuiContext<T>(WrapperComponent: ComponentType<T>): (T) => Node {
  return (props) => (
    <MuiThemeProvider muiTheme={copartBaseTheme}>
      <WrapperComponent {...props} />
    </MuiThemeProvider>
  )
}
