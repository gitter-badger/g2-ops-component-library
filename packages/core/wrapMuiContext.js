// @flow
import type { ComponentType, Node } from 'react'

import React, { Component } from 'react'
import injectTapEventPlugin from 'react-tap-event-plugin'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
import getMuiTheme from 'material-ui/styles/getMuiTheme'
import lightBaseTheme from 'material-ui/styles/baseThemes/lightBaseTheme'
import { initializeIcons } from '@uifabric/icons'

type ThemeType = {
  [string]: any,
}

export const copartBaseTheme: ThemeType = {
  ...lightBaseTheme,
  palette: {
    ...lightBaseTheme.palette,
    primary1Color: '#1d5ab9',
    primary2Color: '#1d5ab9',
  },
  datePicker: {
    ...lightBaseTheme.datePicker,
    headerColor: '#1d5ab9',
  },
  timePicker: {
    ...lightBaseTheme.timepicker,
    headerColor: '#1d5ab9',
  },
  tabs: {
    selectedTextColor: 'rgba(38, 166, 91, 1.0)',
    textColor: 'rgba(218, 223, 225, 1.0)',
  },
  inkBar: {
    backgroundColor: '#545A63',
  },
}

initializeIcons()
injectTapEventPlugin()

// a higher order function that provides mui context to the Component being passed.
export function wrapMuiContext<T>(WrapperComponent: ComponentType<T>): (T) => Node {
  return (props) => (
    <MuiThemeProvider muiTheme={getMuiTheme(copartBaseTheme)}>
      <WrapperComponent {...props} />
    </MuiThemeProvider>
  )
}
