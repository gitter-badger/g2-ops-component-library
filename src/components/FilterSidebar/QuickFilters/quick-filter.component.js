// @flow

import type { QuickFilterType } from 'types/Filter'

import React from 'react'
import cn from 'classnames'
import ReactTooltip from 'react-tooltip'

import './quick-filter.component.scss'

type QuickFilterPropType = {
  quickFilter: QuickFilterType,
  onQuickFiltersChange: string => void,
  isSelected: boolean,
}

const QuickFilter = ({ quickFilter, onQuickFiltersChange, isSelected = false }: QuickFilterPropType) => (
  <div
    data-tip={quickFilter.tooltipText}
    className={cn('QuickFilter', { active: isSelected })}
    onClick={() => onQuickFiltersChange(quickFilter.name)}
  >
    <ReactTooltip />
    <div className="icon">{quickFilter.icon}</div>
    <div className="count">{quickFilter.count}</div>
  </div>
)

export default QuickFilter
