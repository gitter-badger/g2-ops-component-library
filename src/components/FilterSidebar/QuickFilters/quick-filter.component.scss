.QuickFilter {
  display: flex;
  flex-direction: column;
  cursor: pointer;
  width: 58px;
  height: 58px;
  align-items: center;
  justify-content: center;
  .icon {
    svg {
      fill: white !important;
    }
  }
  .count {
    color: #60D1FF;
    font-size: 12px;
  }
  &.active {
    .icon {
      svg {
        fill: #323a49 !important;
      }
    }
    background-color: white;
  }
}
