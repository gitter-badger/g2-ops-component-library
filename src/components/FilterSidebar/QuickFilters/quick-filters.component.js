// @flow

import type { QuickFilterType } from 'types/Filter'

import React, { Component } from 'react'

import QuickFilter from './quick-filter.component'

type QuickFiltersPropType = {
  quickFilters: Array<QuickFilterType>,
  selectedQuickFilter: string,
  onQuickFiltersChange?: (string) => void,
}

type QuickFiltersStateType = {
  selectedQuickFilter: string,
}

class QuickFilters extends Component<QuickFiltersPropType, QuickFiltersStateType> {
  static defaultProps = {
    quickFilters: [],
  }

  state = { selectedQuickFilter: '' }

  componentWillMount() {
    const { selectedQuickFilter } = this.props
    this.setState(() => ({ selectedQuickFilter }))
  }

  componentWillReceiveProps(nextProps: QuickFiltersPropType) {
    if (nextProps.selectedQuickFilter !== this.props.selectedQuickFilter) {
      this.setState(() => ({ selectedQuickFilter: nextProps.selectedQuickFilter }))
    }
  }

  handleQuickFilterChange = (quickFilter: string) => {
    this.setState(() => ({ selectedQuickFilter: quickFilter }))
    if (this.props.onQuickFiltersChange) {
      this.props.onQuickFiltersChange(quickFilter)
    }
  }

  render() {
    const { quickFilters } = this.props
    const { selectedQuickFilter } = this.state
    return (
      <div className="QuickFilters">
        {quickFilters.map((quickFilter) => (
          <QuickFilter
            quickFilter={quickFilter}
            onQuickFiltersChange={this.handleQuickFilterChange}
            isSelected={selectedQuickFilter === quickFilter.name}
          />
        ))}
      </div>
    )
  }
}

export default QuickFilters
