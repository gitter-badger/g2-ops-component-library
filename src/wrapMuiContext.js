import React from 'react'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
import getMuiTheme from 'material-ui/styles/getMuiTheme'
import lightBaseTheme from 'material-ui/styles/baseThemes/lightBaseTheme'
import injectTapEventPlugin from 'react-tap-event-plugin'
import { initializeIcons } from '@uifabric/icons'

const copartBaseTheme = {
  ...lightBaseTheme,
  palette: {
    ...lightBaseTheme.palette,
    primary1Color: '#1d5ab9',
    primary2Color: '#1d5ab9'
  },
  timePicker: {
    ...lightBaseTheme.timepicker,
    headerColor: '#1d5ab9'
  }
}

initializeIcons();
injectTapEventPlugin();

// a higher order function that provides mui context to the Component being passed.
export const wrapMuiContext = (Component) => {
  return class extends React.Component {
    render() {
      return (
        <MuiThemeProvider muiTheme={getMuiTheme(copartBaseTheme)}>
          <Component {...this.props} />
        </MuiThemeProvider>
      )
    }
  }
}
