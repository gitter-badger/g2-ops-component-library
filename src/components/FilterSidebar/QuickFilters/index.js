// @flow
import QuickFilters from './quick-filters.component'

export default QuickFilters
