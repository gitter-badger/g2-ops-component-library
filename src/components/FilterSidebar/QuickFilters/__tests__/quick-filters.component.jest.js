// @flow

import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Car from 'material-ui/svg-icons/maps/directions-car'

import { wrapMuiContext, copartBaseTheme } from '../../../../wrapMuiContext'

import QuickFilters from '../quick-filters.component'

describe('<QuickFilters />', () => {
  let props
  beforeEach(() => {
    props = {
      quickFilters: [
        {
          name: 'quickFilterOne',
          label: 'Quick Filter One',
          count: 200,
          icon: <Car />,
        },
        {
          name: 'quickFilterTwo',
          label: 'Quick Filter',
          count: 200,
          icon: <Car />,
        },
      ],
      selectedQuickFilter: 'quickFilterOne',
      onQuickFiltersChange: jest.fn(),
    }
  })

  test('<QuickFilters /> renders properly', () => {
    const tree = renderer.create(wrapMuiContext(<QuickFilters {...props} />))
    expect(tree.toJSON()).toMatchSnapshot()
  })

  test('<QuickFilters /> should change active filter', () => {
    const tree = shallow(<QuickFilters {...props} />, { context: { muiTheme: copartBaseTheme } })
    expect(tree.state('selectedQuickFilter')).toBe('quickFilterOne')
    expect(
      tree
        .find('QuickFilter')
        .at(1)
        .prop('onQuickFiltersChange')('quickFilterTwo')
    )
    expect(tree.state('selectedQuickFilter')).toBe('quickFilterTwo')
  })
})
