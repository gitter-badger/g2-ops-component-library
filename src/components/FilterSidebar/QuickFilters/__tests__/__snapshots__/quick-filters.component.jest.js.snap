// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuickFilters /> <QuickFilters /> renders properly 1`] = `
<div
  className="QuickFilters"
>
  <div
    className="QuickFilter active"
    data-tip={undefined}
    onClick={[Function]}
  >
    <div
      className="__react_component_tooltip place-top type-dark "
      data-id="tooltip"
    >
      
    </div>
    <div
      className="icon"
    >
      <svg
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fill": "currentColor",
            "height": 24,
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "width": 24,
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M18.92 6.01C18.72 5.42 18.16 5 17.5 5h-11c-.66 0-1.21.42-1.42 1.01L3 12v8c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1h12v1c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-8l-2.08-5.99zM6.5 16c-.83 0-1.5-.67-1.5-1.5S5.67 13 6.5 13s1.5.67 1.5 1.5S7.33 16 6.5 16zm11 0c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zM5 11l1.5-4.5h11L19 11H5z"
        />
      </svg>
    </div>
    <div
      className="count"
    >
      200
    </div>
  </div>
  <div
    className="QuickFilter"
    data-tip={undefined}
    onClick={[Function]}
  >
    <div
      className="__react_component_tooltip place-top type-dark "
      data-id="tooltip"
    >
      
    </div>
    <div
      className="icon"
    >
      <svg
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fill": "currentColor",
            "height": 24,
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "width": 24,
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="M18.92 6.01C18.72 5.42 18.16 5 17.5 5h-11c-.66 0-1.21.42-1.42 1.01L3 12v8c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1h12v1c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-8l-2.08-5.99zM6.5 16c-.83 0-1.5-.67-1.5-1.5S5.67 13 6.5 13s1.5.67 1.5 1.5S7.33 16 6.5 16zm11 0c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5zM5 11l1.5-4.5h11L19 11H5z"
        />
      </svg>
    </div>
    <div
      className="count"
    >
      200
    </div>
  </div>
</div>
`;
